File: src/codes/Users_Manual/emulator/emulator_ultrasound.py
from pyautd3 import AUTD3, Duration, EmitIntensity, Phase, Silencer, Static, Uniform
from pyautd3_emulator import Emulator, Recorder

with Emulator([AUTD3()]) as emulator:

    def f(autd: Recorder) -> None:
        autd.send(Silencer.disable())
        autd.send((Static(), Uniform(phase=Phase(0x40), intensity=EmitIntensity.MAX)))
        autd.tick(Duration.from_millis(1))

    record = emulator.record(f)

    df = record.output_ultrasound()
    print(df)

File: src/codes/Users_Manual/emulator/emulator_ultrasound.rs
# use std::time::Duration;
# use autd3::prelude::*;
use autd3_emulator::*;

# #[tokio::main]
# async fn main() -> Result<(), Box<dyn std::error::Error>> {
let emulator = Controller::builder([AUTD3::new(Vector3::zeros())]).into_emulator();

let record = emulator
    .record(|mut autd| async {
        autd.send(Silencer::disable()).await?;
        autd.send((
            Static::with_intensity(0xFF),
            Uniform::new((Phase::new(0x40), EmitIntensity::new(0xFF))),
        ))
        .await?;
        autd.tick(Duration::from_millis(1))?;
        Ok(autd)
    })
    .await?;

let df = record.output_ultrasound();
dbg!(df);
#     Ok(())
# }